scripts/hera_mini_snap_data_catcher.py

```diff
@@ -0,0 +1,299 @@
+while True:
+    try:
+        corr_cm = HeraCorrCM(redishost=args.redishost)
+        logger.info("HERA CorrCM Redis connection established.")
+
+        mapping = read_maps_from_redis()
+        snap_to_ant_mapping = mapping["snap_to_ant"]
+        if mapping is None:
+            raise ValueError(
+                "None value encountered for antenna/snap mappings in redis."
+            )
+
+        last_time_mapping = {}
+        last_loop_completion = Time.now()
+
+        logger.info("Beginning data catching.")
+        while True:
+            try:
+
+                # use future_array_shapes
+                # Nblts, Nfreqs, Npols
+                # we don't necessarily know how many freqs we'll have.
+                # it is most likely 1024 but this will guard against X-Engine changes.
+                data_array = None
+                # Nblts
+                time_array = []
+                ant_array = []
+
+                downtime = 0
+                file_len = 0
+
+                while downtime < args.max_downtime and file_len < args.max_file_len:
+                    time.sleep(0.01)
+                    # do the thing
+                    # get integrations from HeraCorrCM.get_snaprf_status()
+                    # compare to the last time for each snap
+
+                    for (
+                        snap_hostname,
+                        status,
+                    ) in corr_cm.get_snaprf_status().items():
+
+                        if status["timestamp"] is None or status["timestamp"] == "None":
+                            continue
+
+                        snap, stream = snap_hostname.split(":")
+                        stream = int(stream)
+
+                        hera_ant_num = snap_to_ant_mapping[snap][stream]
+                        if hera_ant_num is None:
+                            continue
+
+                        # arbitrarily assign east as polarization 0
+                        pol = hera_ant_num[-1].lower()
+                        pol_ind = 0 if pol == "e" else 1
+
+                        # trim off the HH and the Polarization
+                        hera_ant_num = int(hera_ant_num[2:-1])
+
+                        timestamp = Time(status["timestamp"]).jd
+
+                        if snap_hostname not in last_time_mapping:
+                            last_time_mapping[snap_hostname] = timestamp
+                        else:
+                            if timestamp <= last_time_mapping[snap_hostname]:
+                                # Time has not progressed yet.
+                                # Just sit tight
+                                continue
+                            last_time_mapping[snap_hostname] = timestamp
+
+                        auto_corr = status["autocorrelation"]
+
+                        if data_array is None:
+                            data_array = np.zeros(
+                                (0, auto_corr.shape[0], 2), dtype=np.float32
```

Comment:
Why is the first axis length 0?

Reply:
The 0 in the shape acts as a placeholder axis so we can accumulate data into it until we reach either timeout condition or a hour of data. I didn't want to necessarily assume how long a data file would get so I let it grow.